pkg/utils/secrets/manager/generate.go

```diff
@@ -278,12 +278,16 @@ func (m *manager) maintainLifetimeLabels(
 	}
 
 	var dataKeyCertificate string
+	var renewAfterValidityPercentage int
 	switch cfg := config.(type) {
 	case *secretsutils.CertificateSecretConfig:
 		dataKeyCertificate = secretsutils.DataKeyCertificate
 		if cfg.CertType == secretsutils.CACert {
 			dataKeyCertificate = secretsutils.DataKeyCertificateCA
 		}
+		if cfg.RenewAfterValidityPercentage != nil {
+			renewAfterValidityPercentage = *cfg.RenewAfterValidityPercentage
+		}
```

Comment:
See above comment: I think you should pass the Percentage from the `GenerateOptions` when calling `maintainLifetimeLabels` and then simply set the according label in the `desiredLabels` map.